pkgs/applications/audio/sunvox/default.nix

```diff
@@ -1,49 +1,88 @@
-{ lib, stdenv, fetchurl, unzip, alsa-lib, libX11, libXi, SDL2 }:
+{ lib
+, stdenv
+, fetchzip
+, alsa-lib
+, autoPatchelfHook
+, libglvnd
+, libjack2
+, libX11
+, libXi
+, makeWrapper
+, SDL2
+}:
 
 let
-  libPath = lib.makeLibraryPath [ stdenv.cc.cc alsa-lib libX11 libXi SDL2 ];
-  arch =
-    if stdenv.isAarch64
-    then "arm64"
-    else if stdenv.isAarch32
-    then "arm_armhf_raspberry_pi"
-    else if stdenv.is64bit
-    then "x86_64"
-    else "x86";
+  platforms = {
+    "x86_64-linux" = "linux_x86_64";
+    "i686-linux" = "linux_x86";
+    "aarch64-linux" = "linux_arm64";
+    "armv7l-linux" = "arm_armhf_raspberry_pi";
+    "x86_64-darwin" = "macos";
+    "aarch64-darwin" = "macos";
+  };
+  bindir = platforms."${stdenv.hostPlatform.system}" or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
 in
 stdenv.mkDerivation rec {
-  pname = "SunVox";
+  pname = "sunvox";
   version = "2.1c";
 
-  src = fetchurl {
+  src = fetchzip {
     url = "https://www.warmplace.ru/soft/sunvox/sunvox-${version}.zip";
-    sha256 = "sha256-yPVcbtlAVbO9uMsFlfZ51T408hA1VPJAI+R+Jdjcyjw=";
+    hash = "sha256-4b9WSzS2UD4yOFDS20r57VtgU7ecapQIDoJtsY29xeg=";
   };
 
-  nativeBuildInputs = [ unzip ];
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    autoPatchelfHook
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    makeWrapper
+  ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    alsa-lib
+    libglvnd
+    libX11
+    libXi
+    SDL2
+  ];
 
-  unpackPhase = "unzip $src";
+  runtimeDependencies = lib.optionals stdenv.hostPlatform.isLinux [
```

Comment:
That is used by autoPatchelfHook, right?

Reply:
[Yes.](https://nixos.org/manual/nixpkgs/stable/#setup-hook-autopatchelfhook)

Reply:
I don't understand the question? It *will* be in the rpath once put into `runtimeDependencies`.